tests/containers/execute_job_in_eks.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Push a container image to the public cloud container registry
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.com>, qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use publiccloud::k8s_utils;
+use mmapi 'get_current_job_id';
+
+sub run {
+    my ($self, $args) = @_;
+
+    $self->select_serial_terminal;
+    install_kubectl();
+
+    my $provider      = $self->provider_factory();
+    my $tag           = join('-', get_var('PUBLIC_CLOUD_RESOURCE_NAME'), get_current_job_id());
+    my $job_name      = $tag =~ s/_/-/r;
+    my $env           = $provider->get_kubectl_env();
+    my $image         = $provider->get_container_image_name($tag);
+    $self->{provider} = $provider;
+    $self->{job_name} = $job_name;
+    $provider->{tag}  = $tag;
+    my $job           = <<EOT;
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: $job_name
+spec:
+  template:
+    spec:
+      containers:
+      - name: main
+        image: $image
+        command: [ "cat", "/etc/os-release" ]
+      restartPolicy: Never
+  backoffLimit: 4
+EOT
```

Comment:
Please place this file into `data/containers` and then use `file_content_replace()` to place the variables.
```perl
file_content_replace("eks_manifest.yaml", IMAGE => $image);
```

I also suggest to modify `eks_apply_manifest()` so it accepts the YAML file name - not directly content.

Reply:
I would only move the file inside of `data/containers` if it is used twice... Otherwise I think it is good practice to keep the infos of *one* test in *one* place.